### ISSUE TITLE
Daily Feed is sending the wrong day to trade bars, causing factor file to factor incorrectly

### ISSUE THREAD

**Issue by jasonrock11** [QuantConnect/Lean#1586]
The the daily file has (please see bold below):
20151231 00:00,487500,492800,485600,485700,12461
**20160104 00:00,480200,481100,464300,471800,227296
20160105 00:00,1234300,1262500,1223100,1258400,22306**
20160107 00:00,1163800,1197400,1149300,1149500,25666

AND the factor file has
20151113,2.551261,1
**20160104,2.569452,1**
20160328,0.969969,1

WHEN i use my start of 20160105 
            SetStartDate(new DateTime(2016, 01, 05););

THEN my first tradbar has a Time of 20160104 with a value of ~47 (consider the factor is using 0.969969 and should be using 0.569452 with a value of ~120.

If I change my start date to 20160104 (or before) the trade bar does us the factor or 0.569452 with a value of ~120.

In other words, if my start date is the same day as a new factor, the trade bar values are wrong; otherwise the are correct.

**Comment by Martin-Molinero** [comment]
Hey @jasonrock11! Sorry for the late response.
Consider the following algorithm:
```
public override void Initialize()
{
    SetStartDate(2016, 06, 17);
    SetEndDate(2016, 06, 18);
    AddEquity("SPY", Resolution.Daily);
}
```
``` factor file
20160317 | 0.929322 | 1 | 203.4
20160616 | 0.934114 | 1 | 207.75
20160915 | 0.93898 | 1 | 213.14
```
The first daily data point starts at 16 and ends on the 17 and by design, it uses the `20160616 | 0.934114` factor file entry see https://github.com/QuantConnect/Lean/blob/master/Common/Data/Auxiliary/FactorFile.cs#L124. Think there is no issue in master.
This might of been an issue that has already been fixed, closing
